js/setup.js

```diff
@@ -46,6 +47,100 @@ chrome.devtools.network.onRequestFinished.addListener(
     requestRenderer.addNetworkRequestEntry(response.request.url);
 });
 
+/**
+ * This function will be injected into the main page and be executed there.
+ * @param {string} extensionId the chrome.runtime.id of the Mp4Inspector
+ */
+function overWriteSourceBufferAppendData(extensionId) {
+  const originalAppendBuffer = SourceBuffer.prototype.appendBuffer;
+  console.log('injecting append buffer listener');
+
+  SourceBuffer.prototype.appendBuffer = function (data) {
+    originalAppendBuffer.call(this, data);
+
+    try {
+      if (!extensionId) {
+        return;
+      }
+
+      chrome.runtime.sendMessage(extensionId, {
+        type: 'segment-appended',
+        data: arrayBufferToBase64(data),
+      });
+    } catch (err) {
+      console.warn(err);
+    }
+  };
+  function arrayBufferToBase64(arrayBuffer) {
```

Comment:
Can we pull that function out? No need to define it inside of `overWriteSourceBufferAppendData`.

Reply:
actually we need it there, since that function alone will get injected. Otherwise i would have to inject the `arrayBufferToBase64` as well

Reply:
Ah ok. All good then! 😌